data-processing-lib/spark/src/data_processing_spark/runtime/spark/spark_launcher.py

```diff
@@ -0,0 +1,219 @@
+import argparse
+import os
+import socket
+from argparse import ArgumentParser
+from typing import Union
+
+import yaml
+from data_processing.data_access import (
+    DataAccess,
+    DataAccessFactory,
+    DataAccessFactoryBase,
+)
+from data_processing.runtime import (
+    AbstractTransformLauncher,
+    TransformExecutionConfiguration,
+)
+from data_processing.utils import get_logger, str2bool
+from data_processing_spark.runtime.spark.runtime_config import (
+    SparkTransformRuntimeConfiguration,
+)
+from data_processing_spark.runtime.spark.spark_execution_config import (
+    SparkExecutionConfiguration,
+)
+from data_processing_spark.runtime.spark.spark_transform import AbstractSparkTransform
+from pyspark.sql import DataFrame, SparkSession
+
+
+logger = get_logger(__name__)
+
+
+class SparkTransformLauncher(AbstractTransformLauncher):
+    """
```

Comment:
Sorry this is not launcher. This is a combination of launcher and executor. Please split this into 2 classes:
1. Something that in Ray is called orchestrator
2. Actual launcher

Reply:
orchestrator is an implementation detail and the "orchestration" is essentially done in side _run_transform() with  spark implementing the "orchestration" piece.  if useful, we can certainly add a separate orchestrator later.

Reply:
and actually, i think the notion of "orchestration" is a ray-specific implementation detail.  That is, our ray implementation has ray worker that is performing the orchestration - very ray specific IMHO.  I don't see why we need to  have an "orchestrator" for python for example?

Reply:
i respectfully disagree